src/main/java/org/spongepowered/common/event/tracking/phase/packet/PacketState.java

```diff
@@ -175,10 +177,16 @@ public boolean spawnEntity(P context, Entity entity, int chunkX, int chunkZ) {
         final ArrayList<Entity> entities = new ArrayList<>(1);
         entities.add(entity);
         try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
-            Sponge.getCauseStackManager().pushCause(player);
-            Sponge.getCauseStackManager().addContext(EventContextKeys.SPAWN_TYPE, SpawnTypes.PLACEMENT);
-            Sponge.getCauseStackManager().addContext(EventContextKeys.NOTIFIER, player);
-            Sponge.getCauseStackManager().addContext(EventContextKeys.OWNER, player);
+            frame.pushCause(player);
+            if (entity instanceof ExperienceOrb) {
```

Comment:
I'm almost tempted to make this based on the entity itself. I'll merge this with some modifications through EntityUtil (so as to unity usage throughout other places the spawn type is determined based on the entity).